### PR TITLE
Add express and angular

### PR DESCRIPTION
**Pull request by darenang44**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/darenang44/throwcast-angular/issues"
+  },
+  "homepage": "https://github.com/darenang44/throwcast-angular#readme",
+  "dependencies": {
+    "angular": "^1.5.8",
+    "body-parser": "^1.15.2",
```
Comment (dting) [inline package.json]: We probably don't need some of these dependencies.
- body-parser - We won't be sending any requests to this server other requests for static files.
- morgan - We don't need to log the requests for static files.
- nodemon - We are only serving static files so we don't need to restart the server.

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html>
+  <script src="../node_modules/angular/angular.min.js"></script>
```
Comment (dting) [inline public/index.html]: We probably don't want to serve our `node_modules` directory.

File: server/server.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+let express = require('express');
+let bodyParser = require('body-parser');
+let morgan = require('morgan');
+let app = express();
+
+app.use(bodyParser.json());
+app.use(morgan('dev'));
+
+app.use(express.static(__dirname + '/public'));
+app.use('/node_modules', express.static('node_modules'));
```
Comment (dting) [inline server/server.js]: See above about not serving our `node_modules` directory.

File: server.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+let express = require('express');
```
Comment (dting) [inline server.js]: Lets use const unless we expect the assignment to be changed.

Comment (dting) [inline server.js]: Actually, since we aren't transpiling, lets just use `var` everywhere.

File: client/index.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html>
+    <title>Throwcast</title>
```
Comment (dting) [inline client/index.html:4]: This looks like it is missing an opening `<head>` tag.

File: .bowerrc [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "directory": "public/lib"
```
Comment (dting) [inline .bowerrc]: If you want the bower components to be installed in `client/lib` this needs to be:

```
{
  "directory": "client/lib"
}
```

File: server.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+let app = express();
+
+app.use(express.static(__dirname + '/client'));
+
+app.listen(8888, function(){
```
Comment (dting) [inline server.js]: I think you want a space after the `function()` and before the `{`:

```
app.listen(8888, function() {
```

Or however you like it, but be consistent.

File: server.js (hunk 4 above)
Comment (darenang44) [inline server.js]: Done

File: client/index.html (hunk 5 above)
Comment (darenang44) [inline client/index.html:4]: Done

File: .bowerrc (hunk 6 above)
Comment (darenang44) [inline .bowerrc]: Done

File: server.js (hunk 7 above)
Comment (darenang44) [inline server.js]: Done

File: client/app/app.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+angular.module('throwcast', [
+  'throwcast.playlist',
+  'throwcast.categories',
+  'throwcast.popular',
+  'throwcast.auth',
+  'ngRoute'
```
Comment (dting) [inline client/app/app.js]: I actually think this will fail since the angular-routes dependency isn't installed.

File: client/app/app.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+])
+.config(function($routeProvider){
```
Comment (dting) [inline client/app/app.js]: So will this config stuff.

File: client/app/app.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+angular.module('throwcast', [
+  'throwcast.playlist',
```
Comment (dting) [inline client/app/app.js]: These modules don't exist yet.

**Comment by dting** [comment]
Let's remove the unused module dependencies, then LGTM

File: client/index.html [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Throwcast</title>
+  </head>
+  <body ng-app>
```
Comment (dting) [inline client/index.html]: We might as well set `ng-app` to `throwcast` now since we've created it.

File: client/app/app.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+      templateUrl: 'app/pod/categories.html',
+      controller: 'CategoriesController'
+  })
+  .when('/playlist', {
+    templateUrl: 'app/pod/playlist.html',
+    controller: 'PlaylistController'
+  })
+  .otherwise({
+    redirectTo: '/'
+  })
```
Comment (dting) [inline client/app/app.js]: Missing semi colon.

File: client/app/app.js (hunk 9 above)
Comment (darenang44) [inline client/app/app.js]: Done

File: client/app/app.js (hunk 10 above)
Comment (darenang44) [inline client/app/app.js]: Done

File: client/index.html (hunk 11 above)
Comment (darenang44) [inline client/index.html]: Done

File: client/app/app.js (hunk 12 above)
Comment (darenang44) [inline client/app/app.js]: Just delete everything since ngRoute has not yet been installed

**Pull request merged**